sdk/src/accounts/pollingUserAccountSubscriber.ts

```diff
@@ -54,97 +56,74 @@ export class PollingUserAccountSubscriber implements UserAccountSubscriber {
 	}
 
 	async addToAccountLoader(): Promise<void> {
-		if (this.accountsToPoll.size > 0) {
+		if (this.callbackId) {
 			return;
 		}
 
-		this.accountsToPoll.set(this.userAccountPublicKey.toString(), {
-			key: 'user',
-			publicKey: this.userAccountPublicKey,
-			eventType: 'userAccountUpdate',
-		});
+		this.callbackId = await this.accountLoader.addAccount(
+			this.userAccountPublicKey,
+			(buffer, slot: number) => {
+				if (!buffer) {
+					return;
+				}
 
-		for (const [_, accountToPoll] of this.accountsToPoll) {
-			accountToPoll.callbackId = await this.accountLoader.addAccount(
-				accountToPoll.publicKey,
-				(buffer, slot) => {
-					if (!buffer) {
-						return;
-					}
-
-					const account = this.program.account[
-						accountToPoll.key
-					].coder.accounts.decode(capitalize(accountToPoll.key), buffer);
-					this[accountToPoll.key] = { data: account, slot };
-					// @ts-ignore
-					this.eventEmitter.emit(accountToPoll.eventType, account);
-					this.eventEmitter.emit('update');
+				if (this.user && this.user.slot >= slot) {
```

Comment:
i think you should use `>` here since account state may be rolled back if commitment != finalized.